src/overlays/actors/ovl_En_Ishi/z_en_ishi.h

```diff
@@ -6,26 +6,40 @@
 struct EnIshi;
 
 typedef void (*EnIshiActionFunc)(struct EnIshi*, PlayState*);
-typedef void (*EnIshiUnkFunc)(struct EnIshi*, PlayState*);
-typedef void (*EnIshiUnkFunc2)(Actor*, PlayState*);
-
-#define ENISHI_GET_1(thisx) ((thisx)->params & 1)
-#define ENISHI_GET_2(thisx) (((thisx)->params >> 1) & 1)
-#define ENISHI_GET_4(thisx) (((thisx)->params >> 2) & 1)
-#define ENISHI_GET_8(thisx) (((thisx)->params >> 3) & 1)
-#define ENISHI_GET_70(thisx) (((thisx)->params >> 4) & 7)
-#define ENISHI_GET_F0(thisx) (((thisx)->params >> 4) & 0xF)
-#define ENISHI_GET_100(thisx) (((thisx)->params >> 8) & 1)
-#define ENISHI_GET_FLAG(thisx) (((thisx)->params >> 9) & 0x7F)
+// used for dust spawn and draw funcs
+typedef void (*EnIshiMultiFunc)(struct EnIshi*, PlayState*);
+typedef void (*EnIshiSpawnDebrisFunc)(Actor*, PlayState*);
```

Comment:
Hmm, the `MultiFuncs` that only spawn dust seem to only use `Actor*` perhaps change those functions to take `Actor*` and than you can have `EnIshiDrawFunc` and `EnishiSpawnFunc`?

Reply:
Changing the types no longer matches?

Reply:
Hmm, do you have a diff of the changes you made? Locally doing what I thought seems to work.

Reply:
I misunderstood the suggestion and was adding supurfluous EnIshi casts back into the new functions, missed that you meant to drop all down to actor reference only.